### DISCUSSION TITLE
Schrödinger update / lost configuration

### DISCUSSION THREAD

**Discussion by Kamil-Radzyminski** [Rem0o/FanControl.Releases#2834] [General]
Hi. A new user here, configured fan control for the first time last week.
Today I noticed popup with a new update, clicked it, cmd opened, performed update, and nothing happened.
Clicked it agin, it did the same thing, but version did not change.
I closed the app, opened it agin and it was 208 (updated) but my configuration was gone (Started to wonder - was that expected? There were no warrnings of such thing.., Would be a bit pissed off after 6 hours of grinding perfect curves to match noise levels of each sepparate fan to certain workloads if i didn't manage to get it back and at least screenshot).
Nevertheless, I closed the app, restarted the PC and 207 opens on startup agin (with old config).
Closing the app, win + FanControl + enter, and the 208 without config opens. 
What is going on? It seems like another version of the app was installed instead of updating existing instance. Should I file an issue ticket, or is this something known/expected?

**Comment by Kamil-Radzyminski** [comment]
I'll mention that it was installed by .net 8.0 installer from release page.
Not sure if thats the best option / can cause unexpected behaviors for some reason?

**Reply by Rem0o** [reply]
Do a search on your PC for FanControl.exe, and see how many pop up. There is clearly two installs going on.

**Reply by Kamil-Radzyminski** [reply]
> Do a search on your PC for FanControl.exe, and see how many pop up. There is clearly two installs going on.

@Rem0o I did a search and indeed there were two instances of this application on my PC.
One in the current use location: "C:\uzytkowe\testing\FanControl\FanControl.exe"
Second one in a location that was the initial install location, which was later deleted (and the app directory was moved to the current location): "C:\użytkowe\testing\FanControl\FanControl.exe"

So I deleted the second location agin, restarted PC, clicked to update.
It created "C:\użytkowe\testing\FanControl\FanControl.exe" agin.

Seems like a path of initial install is stored somewhere which causes to create a new instance of the app in the initial location?
I guess that uninstall and a clean install to the new location will resolve this issue, but it seems to be a minor bug somewhere

**Reply by Rem0o** [reply]
The updater downloads in the current folder, there is nothing more to it.

**Reply by Kamil-Radzyminski** [reply]
It doesn't in my case.
The difference in the directories is a single polish sign: ż -> z, easy to overlook.

Current folder is "C:\uzytkowe\testing\FanControl\FanControl.exe"
Directory, that is used by updater is no longer existing "C:\użytkowe\testing\FanControl\FanControl.exe"
It creates this directory and installs application there.

Screens of the process:
1. C directory without "użytkowe" (just uzytkowe). App is in "C:\uzytkowe\testing\FanControl\FanControl.exe"
![image](https://github.com/user-attachments/assets/db6f43e4-1133-4857-a4e9-4454ed8d045d)
2. I am clicking the update on running app:
![image](https://github.com/user-attachments/assets/89240d45-f731-4440-8768-634338585eb6)
![image](https://github.com/user-attachments/assets/23781166-a574-4e86-acb4-483b73159f13)
Popup with update stays in place, version 207 is kept.
3. C directory has newly created 'użytkowe' directory, with 209 installed there.
![image](https://github.com/user-attachments/assets/97d3d0f2-1a02-42d9-b6f0-1d05074906fa)

The directory that is created is the one that application was initially installed in, but later moved to another one and deleted.

**Reply by Rem0o** [reply]
Wait... the installer does not install in C:\Program Files\FanControl? Or did you run the installer and changed the path in there, and now you get this bug? 

This has to be a bug with the polish character messing with the path.  Never seen that before. 

What I would do is stick the to the default install path of the installer, aka C:\program files\.... and leave it there to avoid further problems.

**Reply by Kamil-Radzyminski** [reply]
The installer allows to choose a specific directory for instalation:
![image](https://github.com/user-attachments/assets/16351285-aefa-4885-a81f-00ac8f60d23d)
I assume these problems are caused by moving the app to another directory after installation.
I performed a fresh install in my chosen directory and will wait for another update to see if the issue persists.
If it does, I'll try installing in the default directory.

**Reply by Rem0o** [reply]
^ Moving the installation folder will definitely cause problems. It will also break the "start on user log on".

**Reply by Kamil-Radzyminski** [reply]
I didn't notice any problems besides update. Start on user log on worked fine.

**Reply by Rem0o** [reply]
The scheduled action needs a start-in folder:
![image](https://github.com/user-attachments/assets/4d3ab37d-49e4-456f-9833-606cc442ab14)


This will not change if you move the folder.
